src/main/java/com/google/sps/servlets/PlanMailServlet.java

```diff
@@ -0,0 +1,217 @@
+    List<DatePair> potentialTimes = getPotentialTimes(freeTimeUtility, timeNeeded);
+
+    PlanMailResponse planMailResponse =
+        new PlanMailResponse(wordCount, averageReadingSpeed, minutesToRead, potentialTimes);
+    // Convert event list to JSON and print to response
+    JsonUtility.sendJson(response, planMailResponse);
+  }
+
+  private List<DatePair> getPotentialTimes(FreeTimeUtility freeTimeUtility, long timeNeeded) {
+    List<DatePair> workFreeInterval = freeTimeUtility.getWorkFreeInterval();
+    List<DatePair> potentialEventTimes = new ArrayList<>();
+    long remainingTime = timeNeeded;
+    for (DatePair interval : workFreeInterval) {
+      Date potentialEnd = new Date(interval.getKey().getTime() + remainingTime);
+      if (interval.getValue().before(potentialEnd)) {
+        remainingTime -= (interval.getValue().getTime() - interval.getKey().getTime());
+        potentialEventTimes.add(interval);
+      } else {
+        potentialEventTimes.add(new DatePair(interval.getKey(), potentialEnd));
+        break;
+      }
+    }
+    return potentialEventTimes;
+  }
+
+  private int getWordCount(GmailClient gmailClient) {
+    GmailClient.MessageFormat messageFormat = GmailClient.MessageFormat.FULL;
+    int numberDays = 7;
+    int wordCount = 0;
+    List<Message> unreadMessages = new ArrayList<>();
+    try {
+      unreadMessages = gmailClient.getUnreadEmailsFromNDays(messageFormat, numberDays);
+    } catch (IOException e) {
+      System.out.println(e);
+    }
+    for (Message message : unreadMessages) {
+      try {
+        wordCount += getMessageSize(message);
+      } catch (MessagingException | IOException e) {
+        System.out.println(e);
+      }
+    }
+    return 104;
+  }
+
+  /**
+   * Get the events in the user's calendars
+   *
+   * @param calendarClient either a mock CalendarClient or a calendarClient with a valid credential
+   * @param timeMin the minimum time to start looking for events
+   * @param timeMax the maximum time to look for events
+   * @return List of Events from all of the user's calendars
+   * @throws IOException if an issue occurs in the method
+   */
+  private List<Event> getEvents(CalendarClient calendarClient, Date timeMin, Date timeMax)
+      throws IOException {
+    List<CalendarListEntry> calendarList = calendarClient.getCalendarList();
+    List<Event> events = new ArrayList<>();
+    for (CalendarListEntry calendar : calendarList) {
+      events.addAll(calendarClient.getUpcomingEvents(calendar, timeMin, timeMax));
+    }
+    return events;
+  }
+
+  /**
+   * Get the events in the user's calendars
+   *
+   * @param calendarClient either a mock CalendarClient or a calendarClient with a valid credential
+   * @return List of Events from all of the user's calendars
+   * @throws IOException if an issue occurs in the method
+   */
+  private List<Event> getEvents(CalendarClient calendarClient) throws IOException {
```

Comment:
This is not currently used - delete if possible

Reply:
`Resolved`